### PR TITLE
implementing getFieldValue (CELDEV-353)

### PR DESCRIPTION
**Pull request by fpichler**
CELDEV-353 : https://synjira.atlassian.net/browse/CELDEV-353

### CODE REVIEW

**Comment by fpichler** [comment]
please review and merge if okay.

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java [hunk 1]
```diff
@@ -633,6 +634,38 @@ private BaseProperty getBaseProperty(BaseObject obj, String key) {
   }
 
   @Override
+  public <T> Optional<T> getFieldValue(XWikiDocument doc, ClassField<T> field) {
+    return Optional.fromNullable(getProperty(checkNotNull(doc), checkNotNull(field)));
```
Comment (msladek) [inline src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:638]: `checkNotNull(doc)` already happens in _getXObjects_ Line 430

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java [hunk 2]
```diff
@@ -633,6 +634,38 @@ private BaseProperty getBaseProperty(BaseObject obj, String key) {
...
+  }
+
+  @Override
+  public <T> Optional<T> getFieldValue(DocumentReference docRef, ClassField<T> field)
+      throws DocumentNotExistsException {
+    return Optional.fromNullable(getProperty(checkNotNull(docRef), checkNotNull(field)));
```
Comment (msladek) [inline src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:644]: `checkNotNull(docRef)` already happens in _getDocumentReadOnly_ Line 136

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java [hunk 3]
```diff
@@ -633,6 +634,38 @@ private BaseProperty getBaseProperty(BaseObject obj, String key) {
...
+    checkNotNull(ignoreValue);
+    Optional<T> property = getFieldValue(docRef, field);
+    if (property.isPresent() && Objects.equal(property.get(), ignoreValue)) {
+      property = Optional.absent();
+    }
+    return property;
+  }
+
+  @Override
   public <T> T getProperty(DocumentReference docRef, ClassField<T> field)
```
Comment (msladek) [inline src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:669]: perhaps we want to discourage using this method in favour of Optional return values by marking it as deprecated?

**Review by msladek — CHANGES_REQUESTED** [review]
thanks for these additions

**Comment by fpichler** [comment]
please fix first https://synjira.atlassian.net/browse/CELDEV-355

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java (hunk 1 above)
Comment (fpichler) [reply src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:638]: This is right. Over five more embedded method calls in the current implementation. Yet, fast failure means checking early.

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java (hunk 2 above)
Comment (fpichler) [reply src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:644]: This is right. Over four more embedded method calls in the current implementation. Yet, fast failure means checking early.

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java (hunk 3 above)
Comment (fpichler) [reply src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:669]: this could be a later discussion. Is not the focus of this issue.

**Review by msladek — APPROVED** [review]

**Pull request merged**